rules-tests/TypeDeclarationDocblocks/Rector/ClassMethod/DocblockReturnArrayFromDirectArrayInstanceRector/Fixture/false_and_true.php.inc

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Rector\Tests\TypeDeclarationDocblocks\Rector\ClassMethod\DocblockReturnArrayFromDirectArrayInstanceRector\Fixture;
+
+function FalseAndTrue()
+{
+    return [
+        'first'  => [true],
+        'second' => [false],
+        'third'  => [true],
+        'fourth' => [false],
+    ];
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclarationDocblocks\Rector\ClassMethod\DocblockReturnArrayFromDirectArrayInstanceRector\Fixture;
+
+/**
+ * @return array<string, true[]|false[]>
```

Comment:
I think in this case, the `bool[]` would be more clear

Reply:
Yes, as I commented before https://github.com/rectorphp/rector-src/pull/7354#issuecomment-3342707026, it can be improved in separate PR